spec/models/source_spec.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Source, type: :model do
+  context 'when saving two Sources with different cases' do
+    it 'saves only one Source' do
```

Comment:
```suggestion
    it 'does NOT create a second source' do
```

Because the test assert `not_to changes`.

Same for the next test, it assert `to change count by 1`, so we cannot describe the test with `saves 2 sources`, instead, we can use `creates a new source`